### ISSUE TITLE
Page(s) warmed up & tracked with the non-ASCII symbols

### ISSUE THREAD

**Issue by Romanavr** [putyourlightson/craft-blitz#730]
## Context:
We have a regular site with Blitz caching enabled. We also have **server-side rewrties** for nginx.
The refresh mode is “_Clear cache, regenerate manually or organically_”.
Also, in the general.php: `limitAutoSlugsToAscii` => `true`,

We have some logic for the vacancy page that needs to get some data from an external source and cache it until a certain date.
For this logic we use the `expiryDate` attribute, which was working fine until we noticed something.

## Problem:
At some point we ran into a problem that the page cache was not being cleared even though the console command was executed and the diagnostic tool was showing the correct expiration date.

Digging more I noticed a strange thing: the diagnostic page for this entry had a French character (or let's just say non-ASCII), while our website is entirely in English. And the slug for this page is set to English in Section/Entry settings.
So, I didn't found any reference in the files for that _french_ slug.
![image](https://github.com/user-attachments/assets/b516efc7-2e0c-4df1-bab1-15339f3ed703)

I also checked the cache entry in the cache table(`blitz_caches`) and found the following.
The page was tracked with the wrong slug.
![image](https://github.com/user-attachments/assets/19e94027-b144-448c-88f5-521a77df121a)

I cleaned cache and visited page with the regular slug. The row were updated with the correct slug.

Then I just tried to reproduce it locally and visited the page locally with the different slugs, for example,
`http://localhost/emplôyment`
`http://localhost/emplôymënt`
![image](https://github.com/user-attachments/assets/49c84f3e-7870-473c-aafe-b11c4619331e)
Each time the track row was updated with different slugs.

## Possible solution
The main problem for me is that because of the wrong slug in the cache table, Blitz tries to clear the cache for the page with the wrong url, however, my main page is 'employment' in plain English, without non-ASCII symbols.

I see that it's possible to visit the page in the same way without Blitz plugin, so I'm not sure should I address this issue to you or not?
At least, if it's possible, could you convert non-ASCII symbols to ASCII when saving cache record? Especially for cases when `limitAutoSlugsToAscii` config set to `true`.
Otherwise, the wrong path is always cleared and the real page is still in the cache.

#### Application Info

- PHP version: 8.2.13
- Craft edition & version: Pro 4.12.0
- Database driver & version: MySQL 8.0.39

#### Installed Plugins

- Amazon S3: 2.2.1
- Blitz: 4.23.0
- Button Box: 4.2.4
- Dumper: 5.0.1
- Embedded Assets: 4.1.2
- Fetch: 2.0.0-beta.1
- Guest Entries: 4.0.1
- ImageOptimize: 4.0.9
- Linkit: 4.0.4.1
- Mailgun: 3.1.0
- Redactor: 3.1.0
- School calendar: 5.0.7
- Sentry Logger: 4.1.6
- SEOmatic: 4.1.2
- Spoon: 4.0.4
- Sprig: 2.13.0
- SuperSort: 4.0.0
- Super Table: 3.0.14
- Tags: 2.0.0
- Vite: 4.0.10

#### Loaded Modules

- codeeditor: nystudio107\codeeditor\CodeEditor
- sprig-core: putyourlightson\sprig\Sprig
- blitz-hints: putyourlightson\blitzhints\BlitzHints
- verbb-base: verbb\base\Base

#### Blitz Plugin Settings

```json
{
    "debug": false,
    "hintsEnabled": true,
    "cachingEnabled": true,
    "refreshCacheEnabled": true,
    "refreshMode": 1,
    "includedUriPatterns": [
        {
            "uriPattern": ".*"
        }
    ],
    "excludedUriPatterns": {
        "1": {
            "uriPattern": "calendar"
        },
        "2": {
            "uriPattern": "search"
        },
        "3": {
            "uriPattern": "article-ajax"
        }
    },
    "cacheStorageType": "putyourlightson\\blitz\\drivers\\storage\\FileStorage",
    "cacheStorageSettings": {
        "folderPath": "@webroot\/cache\/blitz",
        "compressCachedValues": ""
    },
    "cacheStorageTypes": [],
    "cacheGeneratorType": "putyourlightson\\blitz\\drivers\\generators\\HttpGenerator",
    "cacheGeneratorSettings": {
        "concurrency": "1"
    },
    "cacheGeneratorTypes": [],
    "customSiteUris": [],
    "cachePurgerType": "putyourlightson\\blitz\\drivers\\purgers\\DummyPurger",
    "cachePurgerSettings": [],
    "cachePurgerTypes": [],
    "deployerType": "putyourlightson\\blitz\\drivers\\deployers\\DummyDeployer",
    "deployerSettings": [],
    "deployerTypes": [],
    "ssiEnabled": false,
    "ssiTagFormat": "<!--#include virtual=\"{uri}\" -->",
    "detectSsiEnabled": true,
    "esiEnabled": false,
    "queryStringCaching": 0,
    "includedQueryStringParams": [
        {
            "queryStringParam": ".*"
        }
    ],
    "excludedQueryStringParams": [
        {
            "queryStringParam": "gclid"
        },
        {
            "queryStringParam": "fbclid"
        }
    ],
    "apiKey": "",
    "generatePagesWithQueryStringParams": true,
    "purgeAssetImagesWhenChanged": true,
    "refreshCacheAutomaticallyForGlobals": true,
    "refreshCacheWhenElementMovedInStructure": true,
    "refreshCacheWhenElementSavedUnchanged": false,
    "refreshCacheWhenElementSavedNotLive": false,
    "cacheNonHtmlResponses": false,
    "trackElements": true,
    "trackElementQueries": true,
    "excludedTrackedElementQueryParams": [],
    "cacheElements": true,
    "cacheElementQueries": true,
    "cacheDuration": null,
    "nonCacheableElementTypes": [],
    "sourceIdAttributes": [],
    "liveStatuses": [],
    "integrations": [
        "putyourlightson\\blitz\\drivers\\integrations\\CommerceIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\FeedMeIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\SeomaticIntegration"
    ],
    "defaultCacheControlHeader": "no-store",
    "cacheControlHeader": "public, s-maxage=31536000, max-age=0",
    "cacheControlHeaderExpired": "public, s-maxage=5, max-age=0",
    "sendPoweredByHeader": true,
    "outputComments": true,
    "refreshCacheJobPriority": 10,
    "driverJobBatchSize": 100,
    "driverJobPriority": 100,
    "queueJobTtr": 3600,
    "maxRetryAttempts": 10,
    "maxUriLength": 255,
    "mutexTimeout": 1,
    "commands": [],
    "injectScriptEvent": "DOMContentLoaded",
    "injectScriptPosition": 3
}
```

#### Recommendations

- ❌ One or more globals exist and refreshCacheAutomaticallyForGlobals is enabled.
- ❌ The Blitz Hints utility is reporting 18 eager-loading opportunities.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but unchanged.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but not live.
- ✅ Image transforms are configured to be generated before page load.
- ✅ The @web alias is explicitly defined.
- ✅ Queue jobs are configured not to run automatically via web requests.
- ✅ The Async Queue plugin is not installed or enabled.
- ✅ The blitz/cache/refresh-expired console command has been executed within the past 24 hours.

#### Site Tracking 

- Tracked Pages: 121
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 3206
	- Solspace\Calendar\Elements\Event: 2,123
	- craft\elements\Entry: 778
	- craft\elements\Asset: 286
	- craft\elements\Tag: 15
	- craft\elements\Category: 2
	- craft\elements\User: 2
- Tracked Element Queries: 28
	- craft\elements\Entry: 25
	- Solspace\Calendar\Elements\Event: 2
	- craft\elements\Tag: 1
- Tracked Tags: 0

**Comment by bencroker** [comment]
This sounds like a Craft issue. If visiting `/emplôymënt` returns a 200 status code, Blitz will cache it. If it showed a 404 error, which is what it sounds like you are expecting it to, Blitz would _not_ cache it.

**Comment by Romanavr** [comment]
> This sounds like a Craft issue. If visiting `/emplôymënt` returns a 200 status code, Blitz will cache it. If it showed a 404 error, which is what it sounds like you are expecting it to, Blitz would _not_ cache it.

Agree. But can we convert all non-ASCII to ASCII in the `uri` using craft helpers before saving the record? So the cache will be purged on the correct path.

I'll create another question in the Craft repo.

It's not a problem even if the page with that uri is cached, the main problem is that Blitz can't clear the cache for the original slug.
Neither automatically via console command, nor with the “`Refresh cache`” button.

You can easily reproduce it on your own website, [https://putyourlightson.com/plugi%C5%84s](https://putyourlightson.com/plugi%C5%84s)

**Comment by bencroker** [comment]
This would be a non-issue if Craft returned a 404 error, so I’d prefer to see what the Craft support team says about this before making any changes to the plugin (which may have side-effects).
